401-react-clickable-grid/src/js/GridBox.jsx

```diff
@@ -0,0 +1,35 @@
+/**
+ * n × n 单元格组件
+ */
+
+import GridBoxColumn from './GridBoxColumn';
+
+var GridBox = React.createClass({
+    getInitialState: function () {
+        return {rowNum: this.props.rowNum};
+    },
+
+    componentWillReceiveProps: function (nextProps) {
```

Comment:
is there anyway to avoid using componentWillReceiveProps? I haven't fully thought it through, but this is such as simple app I don't feel like we need to use this